dataflow/src/ops/join.rs

```diff
@@ -431,44 +431,44 @@ impl Ingredient for Join {
                     while other_rows.peek().is_some() {
                         if let Some(false) = make_null {
                             // we need to generate a -NULL for all these lefts
-                            ret.push((self.generate_null(other), false).into());
+                            ret.push((self.generate_null(&other), false).into());
```

Comment:
Wouldn't it be better to have the various `generate_*` things take `Cow`s now so that they can just move out the `DataType`s rather than cloning them?